gammapy/scripts/cta_sensitivity.py

```diff
@@ -0,0 +1,289 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+import logging
+import numpy as np
+from astropy.table import Table, Column
+import astropy.units as u
+from ..utils.scripts import get_parser, set_up_logging_from_args
+from ..stats import significance_on_off
+from ..spectrum.models import PowerLaw
+from ..spectrum.utils import CountsPredictor
+from ..scripts import CTAPerf
+
+log = logging.getLogger(__name__)
+
+__all__ = ['SensiEstimator']
+
+
+class SensiEstimator(object):
+    """Estimate differential sensitivity.
+
+    Parameters
+    ----------
+    irf : `~gammapy.scripts.CTAPerf`
+        IRF object
+    livetime : `~astropy.units.Quantity`
+        Livetime (object with the units of time), e.g. 5*u.h
+    slope : `float`, optional
+        Index of the spectral shape (Power-law), should be positive (>0)
+    alpha : `float`, optional
+        On/OFF normalisation
+    sigma : `float`, optional
+        Minimum significance
+    gamma_min : `float`, optional
+        Minimum number of gamma-rays
+    bkg_sys : `float`, optional
+        Fraction of Background systematics relative to the number of ON counts
+
+    Examples
+    --------
+
+    Compute and plot a sensitivity curve for CTA::
+
+        from gammapy.scripts import CTAPerf, SensiEstimator
+        filename = '$GAMMAPY_EXTRA/datasets/cta/CTA-Performance-North-20170327/CTA-Performance-North-20deg-average-30m_20170327.fits'
+        irf = CTAPerf.read(filename)
+        sens = SensiEstimator(irf=irf, livetime='0.5 hour')
+        sens.run()
+        sens.print_results()
+        sens.plot()
+
+    Further examples in :gp-extra-notebook:`cta_sensitivity` .
+
+    Notes
+    -----
+
+    For the moment, only the differential point-like sensitivity is computed at a fixed offset.
+
+    TODO:
+
+    - make the computation for any source size
+    - compute the integral sensitivity
+    - Add options to use different spectral shape?
+    """
+
+    def __init__(self, irf, livetime, slope=2., alpha=0.2, sigma=5., gamma_min=10., bkg_sys=0.05):
+        self.irf = irf
+        self.livetime = u.Quantity(livetime).to('s')
+        self.slope = slope
+        self.alpha = alpha
+        self.sigma = sigma
+        self.gamma_min = gamma_min
+        self.bkg_sys = bkg_sys
+
+        self.energy = None
+        self.diff_sens = None
+
+    def get_bkg(self, bkg_rate):
+        """Return the Background rate for each energy bin
+
+        Parameters
+        ----------
+        bkg_rate : `~gammapy.scripts.BgRateTable`
+            Table of background rate
+
+        Returns
+        -------
+        rate : `~numpy.ndarray`
+            Return an array of background rate (bins in reconstructed energy)
+        """
+        bkg = bkg_rate.data.data * self.livetime
+        return bkg.value * u.Unit('')
```

Comment:
This is returning a rate according to the docstring. Should this be unitless then?